chimg/image_customizer.py

```diff
@@ -0,0 +1,241 @@
+import os
+import subprocess
+import shutil
+from pathlib import Path
+import logging
+from argparse import Namespace
+import sys
+from chimg.chroot import chrootfs_entry_point
+
+logger = logging.getLogger(__name__)
+
+
+class ImageCustomizer:
+    def __init__(
+        self,
+        *,
+        input_image_file: str | Path,
+        output_image_path: str | Path,
+        target_mount_point: str | Path,
+        chimg_config_file: str | Path,
+        overwrite: bool = False,
+    ):
+        """
+        Initialize the ImageCustomizer with the input image file, output image path, and target mount point."
+
+        Args:
+            input_image_file (str | Path): The path to the input image file.
+            output_image_path (str | Path): The path where the modified image will be saved.
+            target_mount_point (str | Path): The mount point for the image.
+            chimg_config_file (str | Path): The path to the chimg config file.
+            overwrite (bool): Whether to overwrite existing output image files.
+        """
+        self.input_image_file = Path(input_image_file)
+        self.output_image_path = Path(output_image_path)
+        self.target_mount_point = Path(target_mount_point)
+        self.chimg_config_file = Path(chimg_config_file)
+        self.overwrite = overwrite
+
+        self.modified_image_file = self.input_image_file.with_suffix(".modifying")
+        self.output_files_name = self.output_image_path.with_suffix("")
+        self.input_image_type = ""
+        self.resolv_conf_existed = False
+
+        # Check if the output image file already exists
+        if os.path.exists(self.output_image_path) and not self.overwrite:
+            raise FileExistsError(
+                f"Error: Output image file '{self.output_image_path}' already exists! Use overwrite=True to overwrite."
+            )
+
+    def _ensure_paths(self):
+        # if the input image is just in the current directory, we dont need to create the directory
+        if os.path.dirname(self.input_image_file) != "":
+            os.makedirs(os.path.dirname(self.modified_image_file), exist_ok=True)
+        os.makedirs(self.target_mount_point, exist_ok=True)
+
+    def _remove_existing_modified_image(self):
+        if os.path.exists(self.modified_image_file):
+            os.remove(self.modified_image_file)
+            logger.info("Removed existing modified image file")
+
+    def _validate_input_image_exists(self):
+        if not os.path.isfile(self.input_image_file):
+            raise FileNotFoundError(f"Error: Input image file '{self.input_image_file}' not found!")
+
+    def _convert_or_copy_image(self):
```

Comment:
can we leave that out and let the user handle this outside of `chimg`? Otherwise we have to add support for other formats, too.